### PR TITLE
Add two examples of aimd

### PR DESCRIPTION
**Pull request by xyxxxxx**

### CODE REVIEW

File: aimd/job/torch_mnist_trainingjob_aimd.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+parser.add_argument('--api_key',
+                    type=str,
+                    help='API Key for requesting AIMD server.')
+parser.add_argument(
+    '--backend',
+    type=str,
+    help='Distributed backend',
+    choices=[dist.Backend.GLOO, dist.Backend.NCCL, dist.Backend.MPI],
+    default=dist.Backend.GLOO)
+parser.add_argument('--host', type=str, help='URL of AIMD server.')
```
Comment (ppkube) [inline aimd/job/torch_mnist_trainingjob_aimd.py:26]: use a more specific argument name. for example, `--aimd_server`

File: aimd/job/torch_mnist_trainingjob_aimd.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        'conv_channels2': 64,
+        'conv_channels3': 64,
+        'conv_kernel_size': 3,
+        'maxpool_size': 2,
+        'linear_features1': 64,
+        'seed': 1,
+    }
+
+    if rank == 0:
+        from t9k import aimd
```
Comment (ppkube) [inline aimd/job/torch_mnist_trainingjob_aimd.py:167]: "Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants..."

https://peps.python.org/pep-0008/#imports

File: aimd/job/job.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      template:
+        spec:
+          securityContext:
+            runAsUser: 1000
+          containers:
+            - name: pytorch
+              command:
+                - python
+                - aimd/job/torch_mnist_trainingjob_aimd.py
+                - "--host"
```
Comment (ppkube) [inline aimd/job/job.yaml:23]: `--aimd_server`

File: aimd/single-device/job.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      template:
+        spec:
+          securityContext:
+            runAsUser: 1000
+          containers:
+            - name: pytorch
+              command:
+                - python
+                - aimd/single-device/torch_mnist_aimd.py
+                - "--host"
```
Comment (ppkube) [inline aimd/single-device/job.yaml:19]: ditto

File: aimd/single-device/torch_mnist_aimd.py [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+from t9k import aimd
+
+parser = argparse.ArgumentParser(
+    description=
+    'Recording of training data of PyTorch model for MNIST with AIMD.')
+parser.add_argument('--api_key',
+                    type=str,
+                    required=True,
+                    help='API Key for requesting AIMD server.')
+parser.add_argument('--host',
```
Comment (ppkube) [inline aimd/single-device/torch_mnist_aimd.py:20]: ditto

File: job/pytorchtrainingjob/ddp/torch_mnist_trainingjob.py [hunk 6]
```diff
@@ -124,6 +124,7 @@ def test(val=False, epoch=None):
     rank = dist.get_rank()
     world_size = dist.get_world_size()
 
+    # rank 0 download datasets in advance
```
Comment (ppkube) [inline job/pytorchtrainingjob/ddp/torch_mnist_trainingjob.py:127]: 1. downloads
2. this comment needs to be moved to right before line 130?

File: aimd/single-device/README.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
+# 单设备训练使用 AIMD
+
+本示例使用 AIMD 记录并展示模型在单个设备上进行的一次训练中的各种数据（以 PyTorch 模型在单个 CPU 或 GPU 上的训练为例）。
```
Comment (ppkube) [inline aimd/single-device/README.md:3]: why '单个设备'? It's single worker training, so “单工作器” is a better?

the sub-dir name `single-device` should be renamed to `singe-worker`?

File: aimd/single-device/job.yaml [hunk 8]
```diff
@@ -0,0 +1,39 @@
+apiVersion: batch.tensorstack.dev/v1beta1
+kind: PyTorchTrainingJob
+metadata:
+  name: torch-mnist-aimd
```
Comment (ppkube) [inline aimd/single-device/job.yaml:4]: use a better name

File: aimd/job/job.yaml [hunk 9]
```diff
@@ -0,0 +1,74 @@
+apiVersion: batch.tensorstack.dev/v1beta1
+kind: PyTorchTrainingJob
+metadata:
+  name: torch-mnist-trainingjob-aimd
```
Comment (ppkube) [inline aimd/job/job.yaml:4]: use a better name to reflect it's a multi-worker setup.

File: aimd/single-device/torch_mnist_aimd.py [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+                logging.info(
+                    'epoch {:d}/{:d}, batch {:5d}/{:d} with loss: {:.4f}'.
+                    format(epoch, epochs, step, steps_per_epoch, train_loss))
+                global_step = (epoch - 1) * steps_per_epoch + step
+
+                trial.log(
+                    metrics_type='train',  # 记录训练指标
+                    metrics={'loss': train_loss},  # 指标名称及相应值
+                    step=global_step,  # 当前全局步数
+                    epoch=epoch)  # 当前回合数
```
Comment (ppkube) [inline aimd/single-device/torch_mnist_aimd.py:81]: these comments are less useful. remove them

File: aimd/single-device/torch_mnist_aimd.py [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+    test_loss = running_loss / len(loader)
+    test_accuracy = correct / len(loader.dataset)
+    msg = '{:s} loss: {:.4f}, {:s} accuracy: {:.4f}'.format(
+        label, test_loss, label, test_accuracy)
+    if val:
+        msg = 'epoch {:d}/{:d} with '.format(epoch, epochs) + msg
+    logging.info(msg)
+
+    trial.log(
+        metrics_type=label,  # 记录验证/测试指标
```
Comment (ppkube) [inline aimd/single-device/torch_mnist_aimd.py:113]: use English for comments

**Review by ppkube — CHANGES_REQUESTED** [review]

File: aimd/job/job.yaml [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+                - "--api_key"
+                - ""
+                - "--log_dir"
+                - "aimd/job/log"
+                - "--no_cuda"
+              workingDir: /t9k/mnt/tutorial-examples
+              image: registry.tensorstack.cn/t9k/pytorch-1.13.0:sdk-0.5.2
+              resources:
+                requests:
+                  cpu: 8000m
```
Comment (ppkube) [inline aimd/job/job.yaml:34]: 8.0 is more readable

File: aimd/job/job.yaml [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+                  memory: 2Gi
+                limits:
+                  cpu: 12000m
```
Comment (ppkube) [inline aimd/job/job.yaml:37]: 12.0

File: aimd/job/torch_mnist_trainingjob_aimd.py [hunk 14]
```diff
@@ -0,0 +1,231 @@
...
+from torch.nn.parallel import DistributedDataParallel as DDP
+import torch.optim as optim
+from torch.utils.tensorboard import SummaryWriter
+from torchvision import datasets, transforms
+
+parser = argparse.ArgumentParser(
+    description='Distributed training of Keras model for MNIST with DDP.')
+parser.add_argument('--api_key',
+                    type=str,
+                    help='API Key for requesting AIMD server.')
```
Comment (ppkube) [inline aimd/job/torch_mnist_trainingjob_aimd.py:19]: communicating with

File: aimd/job/torch_mnist_trainingjob_aimd.py [hunk 15]
```diff
@@ -0,0 +1,231 @@
...
+        'conv_kernel_size': 3,
+        'maxpool_size': 2,
+        'linear_features1': 64,
+        'seed': 1,
+    }
+
+    if rank == 0:
+        from t9k import aimd
+        trial = aimd.create_trial(trial_name='mnist_torch_distributed',
+                                  folder_path='aimd-example')
```
Comment (ppkube) [inline aimd/job/torch_mnist_trainingjob_aimd.py:169]: Please document in the README that  `folder_path` and `trial_name` are hardcoded in the code ... and where to look for the result in the AIMD UI

**Review by ppkube — CHANGES_REQUESTED** [review]

File: aimd/job/torch_mnist_trainingjob_aimd.py (hunk 1 above)
Comment (xyxxxxx) [reply aimd/job/torch_mnist_trainingjob_aimd.py]: changed to `--aimd_host`

File: aimd/job/torch_mnist_trainingjob_aimd.py (hunk 2 above)
Comment (xyxxxxx) [reply aimd/job/torch_mnist_trainingjob_aimd.py]: moved to top

File: aimd/job/job.yaml (hunk 3 above)
Comment (xyxxxxx) [reply aimd/job/job.yaml]: changed to `--aimd_host`

File: aimd/single-device/job.yaml (hunk 4 above)
Comment (xyxxxxx) [reply aimd/single-device/job.yaml]: ditto

File: aimd/single-device/torch_mnist_aimd.py (hunk 5 above)
Comment (xyxxxxx) [reply aimd/single-device/torch_mnist_aimd.py]: ditto

File: job/pytorchtrainingjob/ddp/torch_mnist_trainingjob.py (hunk 6 above)
Comment (xyxxxxx) [reply job/pytorchtrainingjob/ddp/torch_mnist_trainingjob.py]: modifed and moved

File: aimd/single-device/README.md (hunk 7 above)
Comment (xyxxxxx) [reply aimd/single-device/README.md]: `single-device` renamed to `single-worker`

File: aimd/single-device/job.yaml (hunk 8 above)
Comment (xyxxxxx) [reply aimd/single-device/job.yaml]: changed to `torch-mnist-single-aimd`

File: aimd/single-device/torch_mnist_aimd.py (hunk 10 above)
Comment (xyxxxxx) [reply aimd/single-device/torch_mnist_aimd.py]: removed

File: aimd/job/job.yaml (hunk 12 above)
Comment (xyxxxxx) [reply aimd/job/job.yaml]: modified

File: aimd/job/job.yaml (hunk 13 above)
Comment (xyxxxxx) [reply aimd/job/job.yaml]: modified

File: aimd/job/torch_mnist_trainingjob_aimd.py (hunk 14 above)
Comment (xyxxxxx) [reply aimd/job/torch_mnist_trainingjob_aimd.py]: modified

File: aimd/job/README.md [hunk 16]
```diff
@@ -0,0 +1,27 @@
+# Job 使用 AIMD
+
+本示例使用 AIMD 在 Job 分布式训练中记录并展示模型的训练数据（以 PyTorch 模型的多工作器同步训练为例）。
+
+切换到当前目录下，编辑文件 `job.yaml`，在命令中补全 AIMD 服务器的地址（第 24 行）以及您的 API Key（第 26 行）。
+
+```shell
+# cd into current directory
+cd ~/tutorial-examples/aimd/distributed
```
Comment (ppkube) [inline aimd/job/README.md:9]: the folder is still `aimd/job/README.md`

**Review by ppkube — APPROVED** [review]
LGTM

**Pull request merged**